__tests__/middleware-test.js

```diff
@@ -0,0 +1,158 @@
+jest.mock('@agoric/casting', () => {
+    return {
+        iterateReverse: jest.fn(),
+        makeLeader: jest.fn(),
+        makeFollower: jest.fn(),
+    };
+});
+
+jest.mock('../src/lib/rpc.js', () => {
+    return {
+        boardSlottingMarshaller: jest.fn(),
+        networkConfig: { rpcAddrs: ["http://127.0.0.1:26657"], chainName: "agoriclocal" },
+        makeAgoricNames: jest.fn()
+    };
+});
+
+jest.mock('../src/lib/wallet.js', () => {
+    return {
+        getCurrent: jest.fn(),
+    };
+});
+
+jest.mock('../src/oracle/helper.js', () => {
+    const originalModule = jest.requireActual("../src/oracle/helper")
+    return {
+        ...originalModule,
+        readVStorage: jest.fn()
+    };
+});
+
+import { readJSONFile, readVStorage } from '../src/oracle/helper.js';
+import { getOffers, checkSubmissionForRound, queryPrice, queryRound, getOraclesInvitations } from '../src/oracle/middleware';
+import { iterateReverse, makeLeader, makeFollower } from '@agoric/casting';
+import { boardSlottingMarshaller, makeAgoricNames } from '../src/lib/rpc.js'
+import { getCurrent } from '../src/lib/wallet.js'
+
+let iterateReverseOutput = readJSONFile("__tests__/mock-objects/followerReversed.json")
+let currentOutput = readJSONFile("__tests__/mock-objects/current.json")
+let agoricNamesOutput = readJSONFile("__tests__/mock-objects/agoricNames.json")
+
+const asyncIterator = {
+    [Symbol.asyncIterator]: function() {
+    let i = 0;
+    return {
+        next: () => {
+        return new Promise(resolve => {
+            if (i < iterateReverseOutput.length) {
+            resolve({ value: iterateReverseOutput[i++], done: false });
+            } else {
+            resolve({ done: true });
+            }
+        });
+        }
+    };
+    }
+};
+
+iterateReverse.mockReturnValue(asyncIterator)
+makeLeader.mockReturnValue({})
+makeFollower.mockReturnValue({})
+boardSlottingMarshaller.mockReturnValue({})
+getCurrent.mockReturnValue(currentOutput)
+makeAgoricNames.mockReturnValue(agoricNamesOutput)
+
+/**
+ * Test for get offers
+ */
+test('calls getOffers to get latest successful submissions', async () => {
+  
+    const offers = await getOffers({});
+  
+    //notice how only the first offer should be calculated because for every offer there will be 3 entries
```

Comment:
doesn't fit 80 char limit

Reply:
Fixed in https://github.com/jacquesvcritien/agoric-cl-middleware/commit/e7d6c899fec472591f9ed9fa8dfad2c660fbb9a5

Reply:
Can you please check that the character limits have been obeyed throughout? In this case I can still see the comment exceeding 80 chars.

Reply:
Going to resolve all other attempts but keeping only this open so that we don't have bloat.

Reply:
This file was not formatted in that commit but got formatted in a later commit https://github.com/jacquesvcritien/agoric-cl-middleware/commit/2a7bc45b4dfcdcca162fb6a799c0d6ccffe75de5